### ISSUE TITLE
[numerics] Unbalanced Sinkhorn UOT runs in exp-domain; add log-domain (log-sum-exp) stabilization

### ISSUE THREAD

**Issue by PeterPonyu** [PeterPonyu/aether-3d#134]
## Problem
The PyTorch unbalanced-Sinkhorn solver runs in the multiplicative (exp) domain. Issue #23 (closed) mitigated fp32 underflow by promoting to float64 when `(-cost/reg) < -80` is detected, but float64 only buys ~9 extra orders of magnitude and still underflows/overflows for small `reg` (sharper transport) or large normalized costs — and `lambda_class = 10` is *added* to the normalized cost, pushing `-cost/reg` very negative. The `1e-12` clamps inside the `u`/`v` updates mask collapse to a near-uniform plan rather than preventing it.

## Location
`src/aether_3d/coupling/uot.py:198-242` (`compute_uot_coupling_pytorch`); kernel at line 216:
```python
K = torch.exp(-solver_cost / reg) * (a_t.unsqueeze(1) * b_t.unsqueeze(0))
...
u = (a_t / torch.clamp(Kv, min=1e-12)) ** fi   # clamp masks underflow
```

## Why it matters
Small-`reg` / strong-class-penalty regimes are exactly where UOT coupling is most informative. Silent kernel collapse yields a near-uniform coupling, so cross-slice trajectories degrade to noise without any error — directly undermining the central UOT contribution.

## Proposed change / experiment
- Implement a log-domain stabilized unbalanced Sinkhorn: keep dual potentials in log space and update with `torch.logsumexp` over the log-kernel `-cost/reg`; only exponentiate the final transport plan.
- Keep a float32 fast path; retire the float64 band-aid from #23 once stabilized.
- Add tests: (a) parity with `ot.sinkhorn_unbalanced` in a benign regime; (b) a small-`reg` (e.g. 0.05) regime where the current exp-domain solver collapses but the log-domain solver returns a non-degenerate plan.

## Acceptance criteria
- [ ] Log-domain solver matches POT within tolerance in a benign regime.
- [ ] Produces a non-degenerate coupling at `reg <= 0.05` where the exp-domain path collapses.
- [ ] Underflow-detection / float64 promotion removed or explicitly justified.
- [ ] Tests added.

## Cross-repo
none (UOT coupling is Aether-specific; not shared with lumina-st).
